python_modules/dagster-graphql/dagster_graphql_tests/graphql/test_partition_backfill.py

```diff
@@ -242,11 +251,33 @@ def _execute_backfill_iteration_with_side_effects(graphql_context, backfill_id):
     """Executes an asset backfill iteration with side effects (i.e. updates run status and bulk action status)."""
     with get_workspace_process_context(graphql_context.instance) as context:
         backfill = graphql_context.instance.get_backfill(backfill_id)
-        list(
-            execute_asset_backfill_iteration(
-                backfill, logging.getLogger("fake_logger"), context, graphql_context.instance
+        evaluation_time = pendulum.now("UTC")
+        tick = graphql_context.instance.create_tick(
+            TickData(
+                instigator_origin_id=_BACKFILL_ORIGIN_ID,
+                instigator_name=backfill_id,
+                instigator_type=InstigatorType.BACKFILL,
+                status=TickStatus.STARTED,
+                timestamp=evaluation_time.timestamp(),
+                selector_id=backfill_id,
             )
         )
+        tick_retention_settings = graphql_context.instance.get_tick_retention_settings(
+            InstigatorType.BACKFILL
+        )
+        logger = logging.getLogger("fake_logger")
+        with BackfillLaunchContext(
```

Comment:
not sure about needing to manually make the BackfillLaunchContext for the test. An alternate option would be to only make the BackfillLaunchContext for asset backfills, and make it in `execute_asset_backfill_iteration`. But that means we could lose some visibility into which backfill is currently being executed. for example, if a job backfill were taking a really long time, the tick table would have no tick as STARTED, so it would be hard to know from the ticks table which backfill is the problem.

It's also unclear to me if making the context in the test is actually a bad thing. none of the other launch contexts (for sensors, amp, etc) are created as parts of tests, but creating it in the test doesn't jump out as a clearly bad idea to me